### PR TITLE
LCM of N elements of an array Issue No. #1821

### PR DESCRIPTION
**Pull request by ritish099**
Fixes # #1821

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Added code to find LCM of N array of elements

#### Languages Used:
- C 
- C++
- JAVA

#### Files Added:
- LCM.c
- LCM.cpp
- LCM.java
Thanks!

### CODE REVIEW

**Comment by ritish099** [comment]
@SakshiShreya kindly review it

**Comment by MastersAbh** [comment]
Please mention the issue number you are trying to fix

**Comment by ritish099** [comment]
Issue number #1821

**Comment by ritish099** [comment]
@MastersAbh check it.

**Comment by jainaman224** [comment]
Please edit the PR description to include issue number.

**Comment by MastersAbh** [comment]
Please edit the description and add the issue number after Fixes #

**Comment by MastersAbh** [comment]
@ritish099 are you working on it ?

**Comment by ritish099** [comment]
Done
@MastersAbh

**Comment by somya-kapoor** [comment]
Please change the description use [x] instead of [Yes].

File: LCM_of_array_of_elements/LCM.c [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  
+    return ans; 
+} 
+  
+// Driver Code 
+int main() 
+{ 
+	int n,i;
+	printf("Enter the no of elements of the array\n");
+	scanf("%d",&n);
```
Comment (somya-kapoor) [inline LCM_of_array_of_elements/LCM.c]: Add space after comma.

File: LCM_of_array_of_elements/LCM.c [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+	int arr[n];
+	printf("Enter the elements of the array\n");
+	for(i=0;i<n;i++)
```
Comment (somya-kapoor) [inline LCM_of_array_of_elements/LCM.c]: Add space around the operator.

File: LCM_of_array_of_elements/LCM.c [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        ans = (((arr[i] * ans)) / 
+                (gcd(arr[i], ans))); 
+  
+    return ans; 
+} 
+  
+// Driver Code 
+int main() 
+{ 
+	int n,i;
```
Comment (somya-kapoor) [inline LCM_of_array_of_elements/LCM.c]: Add space after comma.

File: LCM_of_array_of_elements/LCM.c [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+	printf("Enter the no of elements of the array\n");
+	scanf("%d",&n);
+	int arr[n];
+	printf("Enter the elements of the array\n");
+	for(i=0;i<n;i++)
+	{
+		scanf("%d",&arr[i]);
```
Comment (somya-kapoor) [inline LCM_of_array_of_elements/LCM.c]: Add space after comma.

File: LCM_of_array_of_elements/LCM.cpp [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+   for(int i = 2; i < n; i++)
+   {
+      lcm = getLCM(lcm, arr[i]); //calls other function getLCM to compute the LCM
+   }
+   return lcm; //returns LCM to main.
+}
+int main() 
+{
+	int n,i;
+	cout<<"Enter the no. of elements of array\n";
```
Comment (somya-kapoor) [inline LCM_of_array_of_elements/LCM.cpp]: Add space around << and >>

File: LCM_of_array_of_elements/LCM.cpp [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+	cin>>n;
+	int a[n];
+	cout<<"Enter the elements of array\n";
+	for(i=0;i<n;i++)
```
Comment (somya-kapoor) [inline LCM_of_array_of_elements/LCM.cpp]: Add space around the operator.

File: LCM_of_array_of_elements/LCM.java [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+        Scanner s = new Scanner(System.in);
+        System.out.println("Enter no. of elements in array:");
+        n = s.nextInt();
+        int a[] = new int[n];
+        System.out.println("Enter all the elements:");
+        for(int i = 0; i < n; i++)
+        {
+            a[i] = s.nextInt();//Taking input from the user            
+        }
+        int lcm=1;
```
Comment (somya-kapoor) [inline LCM_of_array_of_elements/LCM.java]: Add space around =.

File: LCM_of_array_of_elements/LCM.java [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+    //Calculating LCM by using GCD
+    private static int calculateLcm(int lcm, int each) 
+    {
+        return lcm*each/gcd(lcm,each);
+    }
+
+    //Calculating GCD
+    private static int gcd(int val1, int val2) 
+    {
+        if(val1==0||val2==0)
```
Comment (somya-kapoor) [inline LCM_of_array_of_elements/LCM.java]: Add space around the operator

File: LCM_of_array_of_elements/LCM.java [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+            return 0;
+
+        if(val1==val2)
```
Comment (somya-kapoor) [inline LCM_of_array_of_elements/LCM.java]: Add space around the operator

File: LCM_of_array_of_elements/LCM.java [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+            return val1;
+
+        if(val1>val2)
```
Comment (somya-kapoor) [inline LCM_of_array_of_elements/LCM.java]: Add space around the operator

**Comment by ritish099** [comment]
@somya-kapoor review it

File: LCM_of_array_of_elements/LCM.c [hunk 11]
```diff
@@ -0,0 +1,45 @@
+// C program to find LCM of n elements
```
Comment (jainaman224) [inline LCM_of_array_of_elements/LCM.c]: Fix file name and folder name to `largest_common_multiple`.

File: LCM_of_array_of_elements/LCM.c [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+#include <stdio.h>
+#include <stdlib.h>
+// GCD of 'a' and 'b'
+int gcd(int a, int b) 
+{
+  if (b == 0)
```
Comment (jainaman224) [inline LCM_of_array_of_elements/LCM.c]: Use 4 space indentation.

File: LCM_of_array_of_elements/LCM.c [hunk 13]
```diff
@@ -0,0 +1,45 @@
...
+  scanf("%d", &n);
+  int arr[n];
+  printf("Enter the elements of the array\n");
+  for (i = 0; i < n; i++) 
+  {
+    scanf("%d", &arr[i]);
+  }
+  printf("LCM : %lld", findlcm(arr, n));
+  return 0;
+}/*Enter the no of elements of the array
```
Comment (jainaman224) [inline LCM_of_array_of_elements/LCM.c]: Move comment to new line.

**Comment by ritish099** [comment]
@asha15 review this one also mam

**Comment by somya-kapoor** [comment]
@ritish099 Please squash the commits, then will be ready to merge.

**Comment by ritish099** [comment]
@somya-kapoor Squashed the commits. Now merge it, please

**Comment by somya-kapoor** [comment]
@ritish099 I just saw the above requests by @jainaman224. You haven't change the file to largest_common_multiple please do that.

File: LCM_of_array_of_elements/LCM.c (hunk 11 above)
Comment (somya-kapoor) [reply LCM_of_array_of_elements/LCM.c]: Please solve this @ritish099

**Comment by asha15** [comment]
@jainaman224  could you approve this PR

File: Largest_Common_Multiple/Largest_Common_Multiple_1-DArray.cpp [hunk 14]
```diff
@@ -0,0 +1,47 @@
+/*This code is to find the Largest__common_multiple of an given array of elements*/ 
+
+#include <iostream>
+
+using namespace std;
+int getLCM(int a, int b)
```
Comment (somya-kapoor) [inline Largest_Common_Multiple/Largest_Common_Multiple_1-DArray.cpp]: Add a line above it

File: Largest_Common_Multiple/Largest_Common_Multiple_1-DArray.cpp [hunk 15]
```diff
@@ -0,0 +1,47 @@
...
+{
+    int m;
+    m = (a > b) ? a : b;
+    while (true) {
+        if (m % a == 0 && m % b == 0)
+            return m; // returns the value to getLCMArray function
+        m++;
+    }
+}
+int getLCMArray(int arr[], int n)
```
Comment (somya-kapoor) [inline Largest_Common_Multiple/Largest_Common_Multiple_1-DArray.cpp]: Add a line above it

File: Largest_Common_Multiple/Largest_Common_Multiple_1-DArray.cpp [hunk 16]
```diff
@@ -0,0 +1,47 @@
...
+int getLCMArray(int arr[], int n) 
+{
+    int lcm = getLCM(arr[0], arr[1]);
+    for (int i = 2; i < n; i++) 
+    {
+        lcm = getLCM(lcm, arr[i]); // calls other function getLCM to compute the LCM
+    }
+    return lcm; // returns LCM to main.
+}
+int main()
```
Comment (somya-kapoor) [inline Largest_Common_Multiple/Largest_Common_Multiple_1-DArray.cpp]: Add a line above it

File: Largest_Common_Multiple/Largest_Common_Multiple_1-DArray.java [hunk 17]
```diff
@@ -0,0 +1,52 @@
+// JAVA code to find Largest_common_multiple of n elements of the array
+
+import java.util.Scanner;
+import java.util.Arrays;
+class LCM
```
Comment (somya-kapoor) [inline Largest_Common_Multiple/Largest_Common_Multiple_1-DArray.java]: Add a line above it

File: Largest_Common_Multiple/Largest_Common_Multiple_1-DArray.c [hunk 18]
```diff
@@ -0,0 +1,48 @@
+// C program to find Largest_common_multiple of n elements
+
+#include <stdio.h>
+
+#include <stdlib.h>
+ // GCD of 'a' and 'b'
```
Comment (somya-kapoor) [inline Largest_Common_Multiple/Largest_Common_Multiple_1-DArray.c]: inconsistency in indentation

File: Largest_Common_Multiple/Largest_Common_Multiple_1-DArray.c [hunk 19]
```diff
@@ -0,0 +1,48 @@
+// C program to find Largest_common_multiple of n elements
+
+#include <stdio.h>
+
```
Comment (somya-kapoor) [inline Largest_Common_Multiple/Largest_Common_Multiple_1-DArray.c]: Remove this line

File: Largest_Common_Multiple/Largest_Common_Multiple_1-DArray.cpp [hunk 20]
```diff
@@ -0,0 +1,47 @@
...
+#include <iostream>
+
+using namespace std;
+int getLCM(int a, int b) 
+{
+    int m;
+    m = (a > b) ? a : b;
+    while (true) {
+        if (m % a == 0 && m % b == 0)
+            return m; // returns the value to getLCMArray function
```
Comment (somya-kapoor) [inline Largest_Common_Multiple/Largest_Common_Multiple_1-DArray.cpp]: move comments above the statements

File: Largest_Common_Multiple/Largest_Common_Multiple_1-DArray.cpp [hunk 21]
```diff
@@ -0,0 +1,47 @@
...
+        m++;
+    }
+}
+int getLCMArray(int arr[], int n) 
+{
+    int lcm = getLCM(arr[0], arr[1]);
+    for (int i = 2; i < n; i++) 
+    {
+        lcm = getLCM(lcm, arr[i]); // calls other function getLCM to compute the LCM
```
Comment (somya-kapoor) [inline Largest_Common_Multiple/Largest_Common_Multiple_1-DArray.cpp]: move comments above the statements

File: Largest_Common_Multiple/Largest_Common_Multiple_1-DArray.cpp [hunk 22]
```diff
@@ -0,0 +1,47 @@
...
+    }
+    return lcm; // returns LCM to main.
```
Comment (somya-kapoor) [inline Largest_Common_Multiple/Largest_Common_Multiple_1-DArray.cpp]: move comments above the statements

File: Largest_Common_Multiple/Largest_Common_Multiple_1-DArray.java [hunk 23]
```diff
@@ -0,0 +1,52 @@
...
+    {
+        int n;
+        Scanner s = new Scanner(System.in);
+        System.out.println("Enter no. of elements in array:");
+        n = s.nextInt();
+        int a[] = new int[n];
+        System.out.println("Enter all the elements:");
+        for (int i = 0; i < n; i++) 
+        {
+            a[i] = s.nextInt(); // Taking input from the user
```
Comment (somya-kapoor) [inline Largest_Common_Multiple/Largest_Common_Multiple_1-DArray.java]: move comments above the statements

File: Largest_Common_Multiple/Largest_Common_Multiple_1-DArray.cpp (hunk 21 above)
Comment (somya-kapoor) [reply Largest_Common_Multiple/Largest_Common_Multiple_1-DArray.cpp]: rewrite the statement

File: Largest_Common_Multiple/Largest_Common_Multiple_1-DArray.c (hunk 18 above)
Comment (somya-kapoor) [inline Largest_Common_Multiple/Largest_Common_Multiple_1-DArray.c]: Add a line above it

Comment (somya-kapoor) [reply Largest_Common_Multiple/Largest_Common_Multiple_1-DArray.c]: Add a line above it

File: Largest_Common_Multiple/Largest_Common_Multiple_1-DArray.java [hunk 24]
```diff
@@ -0,0 +1,55 @@
+// JAVA code to find Largest_common_multiple of n elements of the array
+
+import java.util.Scanner;
+
```
Comment (somya-kapoor) [inline Largest_Common_Multiple/Largest_Common_Multiple_1-DArray.java]: remove this line

**Pull request merged**